### PR TITLE
Dev faceseg

### PR DESCRIPTION
**Pull request by JiaojiaoYe1994**
rectify dependency in requirement.txt

### CODE REVIEW

File: FaceSeg/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+This repo is about face segmentation based on LinkNet34, our work is inspired by this PyTorch [implementaion](https://github.com/JiaojiaoYe1994/face-segmentation).
+
+## Environment
+| Spec                        |                                                             |
+|-----------------------------|-------------------------------------------------------------|
+| Operating System            | Ubuntu 18.04                                        |
+| GPU                         | Nvidia A100-SXM4-40GB                          |
+| CUDA Version                | 11.0                                                   |
+| Driver Version              | 460.73.01                                             |
+| Oneflow Version 	          | branch: master, commit_id: 90d3277a098f483d0a0e68621b7c8fb2497a6fc2 |
```
Comment (rentainhe) [inline FaceSeg/README.md:12]: 对齐一下这个issue的oneflow版本 https://github.com/Oneflow-Inc/models/issues/149 然后重新测试一下
- `infer.sh` 可以正常执行输出结果
- `train.sh` 可以正常运行并且正常收敛
- 确保这些行为和README一致

File: FaceSeg/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+We combine several dataset and preprocess them, for training and evaluating our 
+face segmentation model. We provide the final compressed dataset, please download dataset from 
+[here](https://oneflow-static.oss-cn-beijing.aliyuncs.com/train_data_zjlab/faceseg_data.zip) , unzip it and put it in `./`
+
+### Run Oneflow Training script
+
+Hyperparameters can be customized, the provided model is trained by run
+
+```bash
+bash train_oneflow.sh
```
Comment (rentainhe) [inline FaceSeg/README.md:49]: 后续把文件里的`train_oneflow.sh`统一改成`train.sh`

File: FaceSeg/models/LinkNet34.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return x
+
+
+class DecoderBlockLinkNet(nn.Module):
+    def __init__(self, in_channels, n_filters) -> None:
+        super().__init__()
+
+        self.relu = nn.ReLU(inplace=True)
+
+        # B, C, H, W -> B, C/4, H, W
```
Comment (rentainhe) [inline FaceSeg/models/LinkNet34.py:31]: 删除这些注释，或者改为以下形式
`(B, C, H, W) -> (B, C/4, H, W)`

File: FaceSeg/models/LinkNet34.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        self.finaldeconv1 = nn.ConvTranspose2d(filters[0], 32, 3, stride=2)
+        self.finalrelu1 = nn.ReLU(inplace=True)
+        self.finalconv2 = nn.Conv2d(32, 32, 3)
+        self.finalrelu2 = nn.ReLU(inplace=True)
+        self.finalconv3 = nn.Conv2d(32, num_classes, 2, padding=1)
+
+    # noinspection PyCallingNonCallable
+    def forward(self, x):
+        # Encoder
+        # print('shape of input is: '+str(x.shape))
```
Comment (rentainhe) [inline FaceSeg/models/LinkNet34.py:95]: 删除注释

File: FaceSeg/train.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    train_aug = Compose([
+        HorizontalFlip(p=0.5),
+        Rotate(15),
+    ])
+
+    train_data_loader = face_seg(
+        args.dataset_path, args.batch_size, augmentation=train_aug)
+    print(len(train_data_loader))
+
+    #################
```
Comment (rentainhe) [inline FaceSeg/train.py:65]: 写一个 #oneflow_init即可，可以把这些`#####`删除

File: FaceSeg/train_oneflow.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
+set -aux
+
+LEARNING_RATE=0.001
+MOM=0.9
+EPOCH=10
+MODEL_PATH="./pretrained_resnet_oneflow_model"
+DATASET_PATH="./faceseg_data/"
+SAVE_MODEL_NAME="linknet_oneflow_model"
+# LOAD PREVIOUS CHECKPOINT
+# LOAD_CHECKPOINT=$CHECKPOINT_PATH/epoch_2_val_acc_0.111168
```
Comment (rentainhe) [inline FaceSeg/train_oneflow.sh:10]: 这里改为 `LOAD_CHECKPOINT="/PATH/TO/CHECKPOINT"`

File: FaceSeg/utils/numpy_data_utils.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import os
+import random
+
+
+def load_image(image_path='data/fish.jpg'):
+    rgb_mean = [123.68, 116.779, 103.939]
+    rgb_std = [58.393, 57.12, 57.375]
+    im = Image.open(image_path)
+    im = im.resize((256, 256))
+    im = im.convert('RGB')  # 有的图像是单通道的，不加转换会报错
```
Comment (rentainhe) [inline FaceSeg/utils/numpy_data_utils.py:13]: 删除注释或者改为英文版本

**Comment by rentainhe** [comment]
可以在评论里把`train` `infer`的结果截图发一下

File: FaceSeg/README.md (hunk 1 above)
Comment (JiaojiaoYe1994) [reply FaceSeg/README.md]: The log file for run bash files.
* train log:
train sh

* infer log:
infer_sh

**Review by rentainhe — APPROVED** [review]
